### ISSUE TITLE
Pyright doesn't report unbound variable when type checking is off

### ISSUE THREAD

**Issue by oyarsa** [microsoft/pyright#4921]
**Describe the bug**
pyright doesn't report a possibly unbound variable with type checking off.

**To Reproduce**
```python
# unbound.py
x = 0
if x > 1:
    y = x - 1
print(y)
```

Running pyright with the default setting shows the following message:

```
❯ pyright unbound.py
No configuration file found.
No pyproject.toml file found.
Assuming Python platform Darwin
Searching for source files
Found 1 source file
pyright 1.1.302
/Users/italo/tmp/unbound/unbound.py
  /Users/italo/tmp/unbound/unbound.py:5:7 - error: "y" is possibly unbound (reportUnboundVariable)
1 error, 0 warnings, 0 informations 
Completed in 0.459sec
```

Turning off type checking makes the error disappear:

```
❯ cat pyrightconfig.json 
{
  "typeCheckingMode": "off"
}

❯ pyright unbound.py
Loading configuration file at /Users/italo/tmp/unbound/pyrightconfig.json
Assuming Python version 3.11
Assuming Python platform Darwin
Auto-excluding **/node_modules
Auto-excluding **/__pycache__
Auto-excluding **/.*
Searching for source files
Found 1 source file
pyright 1.1.302
0 errors, 0 warnings, 0 informations 
Completed in 0.467sec
```

Explicitly turning on the error doesn't change anything:

```
❯ cat pyrightconfig.json
{
  "typeCheckingMode": "off",
  "reportUnboundVariables": "on"
}⏎                                                                                                       

❯ pyright unbound.py
Loading configuration file at /Users/italo/tmp/unbound/pyrightconfig.json
Assuming Python version 3.11
Assuming Python platform Darwin
Auto-excluding **/node_modules
Auto-excluding **/__pycache__
Auto-excluding **/.*
Searching for source files
Found 1 source file
pyright 1.1.302
0 errors, 0 warnings, 0 informations 
Completed in 0.464sec
```

**Expected behavior**
pyright with type checking off should still report the possibly unbound error.

**VS Code extension or command-line**
The same issue happens on both command-line and within VS Code with Pylance.

**Additional context**
Python version: 3.11.2 (installed via pyenv)
Pyright version: 1.1.302 (installed via pip)
OS:  macOS 13.2.1 (Apple Silicon)

Examples ran in a clean virtual environment.

**Comment by erictraut** [comment]
Yes, this is working as designed. Unbound variables are detected as part of type checking. If you disable all type checking errors, you will not see unbound variables reported.

**Comment by erictraut** [comment]
Note that you can enable the `reportUnboundVariable` check independent of `typeCheckingMode`.

**Comment by oyarsa** [comment]
That doesn't work:

``` 
❯ cat pyrightconfig.json
{
  "typeCheckingMode": "off",
  "reportUnboundVariable": "on"
} 
 
❯ pyright unbound.py
Loading configuration file at /Users/italo/tmp/pyrightconfig.json
Config "reportUnboundVariable" entry must be true, false, "error", "warning", "information" or "none".
Assuming Python version 3.11
Assuming Python platform Darwin
Auto-excluding **/node_modules
Auto-excluding **/__pycache__
Auto-excluding **/.*
Searching for source files
Found 1 source file
pyright 1.1.302
0 errors, 0 warnings, 0 informations 
Completed in 0.459sec
```

**Comment by oyarsa** [comment]
I thought I'd covered that, but I was setting it wrong. This works, thank you.
